### ISSUE TITLE
I have a question that how to change learning rate and epoch

### ISSUE THREAD

**Issue by HuangZhanPeng** [cnclabs/proNet-core#27]
Dose this library provide the hyper parameter learning rate, epoch and other optimize algorithm
Because I do not know the training is coverge

I also has one more question:

the bpr algorithm input format is

user pos_item neg_item?

**Comment by chihming** [comment]
In the previous commit, I've added the reg parameter for training. Hence, there are: **dimensions, sample_times, reg, alpha**. 4 hyperparameters

**Comment by HuangZhanPeng** [comment]
thank you for sharing and answer my question!!! looking forword more excellent update
